### PR TITLE
Ensure `sudo` both work in powershell and pwsh.

### PR DESCRIPTION
**Pull request by zhoujin7**
The problem is solved. #20 
Thanks @jtc42 for providing a workaround of pwsh.
[https://github.com/jtc42/psutils/blob/master/sudo.ps1](https://github.com/jtc42/psutils/blob/master/sudo.ps1)
Now,  `sudo` both work in powershell and pwsh.

### CODE REVIEW

**Comment by 573** [comment]
Would like to see this merged.

File: sudo.ps1 [hunk 1]
```diff
@@ -22,8 +27,8 @@ function sudo_do($parent_pid, $dir, $cmd) {
 	$kernel::attachconsole($parent_pid)
 		
 	$p = new-object diagnostics.process; $start = $p.startinfo
-	$start.filename = "powershell.exe"
-	$start.arguments = "-noprofile $cmd`nexit `$lastexitcode"
+	$start.filename = "$powershellExe"
+	$start.arguments = "-noprofile -Command $cmd`nexit `$lastexitcode"
```
Comment (573) [inline sudo.ps1]: You could use $commandPrefix or similar here as well such that major-versions less equals 5 ain't broken:

    	$start.arguments = "-noprofile $commandPrefix $cmd`nexit `$lastexitcode"

File: sudo.ps1 [hunk 2]
```diff
@@ -2,6 +2,11 @@ Set-StrictMode -Off;
 
 if(!$args) { "usage: sudo <cmd...>"; exit 1 }
 
+$powershellExe = 'powershell.exe'
+if ($host.Version.Major -gt 5) {
+  $powershellExe = 'pwsh.exe'
+}
+
```
Comment (573) [inline sudo.ps1]: $commandPrefix = ''
    if ($host.Version.Major -gt 5) {
        $commandPrefix = '-Command'
    }

**Comment by jtc42** [comment]
Oh this is a much nicer way of doing it than mine! Thanks for this.

File: sudo.ps1 [hunk 3]
```diff
@@ -61,9 +66,10 @@ $wd = serialize (convert-path $pwd) # convert-path in case pwd is a PSDrive
 
 $savetitle = $host.ui.rawui.windowtitle
 $p = new-object diagnostics.process; $start = $p.startinfo
-$start.filename = "powershell.exe"
-$start.arguments = "-noprofile & '$pscommandpath' -do $wd $pid $a`nexit `$lastexitcode"
+$start.filename = "$powershellExe"
+$start.arguments = "-noprofile -Command & '$pscommandpath' -do $wd $pid $a`nexit `$lastexitcode"
```
Comment (573) [inline sudo.ps1]: Same as said in l. 31:

    $start.arguments = "-noprofile $commandPrefix & '$pscommandpath' -do $wd $pid $a`nexit `$lastexitcode"

File: sudo.ps1 (hunk 1 above)
Comment (zhoujin7) [reply sudo.ps1]: Thanks for your suggestions.
But scoop seems to support powershell major version greater than 5 only.

Comment (573) [reply sudo.ps1]: Sorry for the confusion on my side, just see ll. 6 in the original sudo.ps1 file to see what I wanted to say.
I definitely mean to support both the `-Command` prefixed and the non-prefixed syntax regardless of major version.

File: sudo.ps1 [hunk 4]
```diff
@@ -2,6 +2,11 @@ Set-StrictMode -Off;
 
 if(!$args) { "usage: sudo <cmd...>"; exit 1 }
 
+$powershellExe = 'powershell.exe'
```
Comment (JustinGrote) [inline sudo.ps1]: Why guess at this? Why not just use (get-process -id $pid).path and instantiate the same shell? This works cross platform and doesn't run the risk that you will fire up a different version pwsh than you intended (for instance if you're running powershell preview in a custom directory)

Then none of this psversiontable comparison stuff needs to be done, it'll just use whatever shell it was invoked from.

Comment (JustinGrote) [reply sudo.ps1]: @lukesampson I'll start a new PR if @zhoujin7 has abandoned this one, if you're interested.

**Pull request closed without merging**